### PR TITLE
Categorize documents in the sidebar of the QGIS documentation website 

### PR DESCRIPTION
**Pull request by muranamihdk**
Categorize documents into three groups of "For users", "For documentation writers" and "For developers" in the sidebar of the QGIS documentation website in accordance with the description of the documentation area in the [QGIS website](https://qgis.org/en/docs/).
https://docs.qgis.org/testing/en/docs/index.html

screenshot2

screenshot

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -9,13 +9,24 @@ Please have a look into one of the documents below.
...
+   :caption: For Users
 
    QGIS Desktop User Guide/Manual (QGIS Testing) <user_manual/index>
    QGIS Server Guide/Manual (QGIS Testing) <server_manual/index>
+   Training Manual <training_manual/index>
+   A Gentle Introduction to GIS <gentle_gis_introduction/index>
+
+.. toctree::
+   :maxdepth: 2
+   :caption: For Documentation Writers
```
Comment (DelazJ) [inline docs/index.rst:21]: Maybe shorten this to "for writers"?

**Review by DelazJ — APPROVED** [review]
Good idea @muranamihdk

File: docs/index.rst (hunk 1 above)
Comment (muranamihdk) [reply docs/index.rst:21]: Or "For Documenters"? Which is suitable?

Comment (muranamihdk) [reply docs/index.rst]: I changed "For Documentation Writers" into "For Writers".

**Comment by DelazJ** [comment]
Thanks @muranamihdk

**Comment by muranamihdk** [comment]
@DelazJ Thank you for your review!
